src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -238,26 +155,38 @@ String getClassNameInternal() {
             int nRSConcur,
             SQLServerStatementColumnEncryptionSetting stmtColEncSetting) throws SQLServerException {
         super(conn, nRSType, nRSConcur, stmtColEncSetting);
+
+        if (null == sql) {
```

Comment:
I am wondering if this change may break existing apps. Not sure how but if we don't throw on "null" SQL text today this may create an issue...